pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -403,7 +396,11 @@ def barrier(self, *args, **kwargs) -> None:
             torch.distributed.barrier()
 
     def broadcast(self, obj: object, src: int = 0) -> object:
-        return self.dist.broadcast(obj)
+        obj = [obj]
```

Comment:
- please address the bug with `src` not being respected
- as discussed elsewhere, we can special case Tensor objects for a potential speedup as @yifuwang noted here: https://github.com/PyTorchLightning/pytorch-lightning/pull/9414#discussion_r707788699

Reply:
>please address the bug with src not being respected

It's address here, as src will be pass into `broadcast_object_list`, before the src was 0

>as discussed elsewhere, we can special case Tensor objects for a potential speedup as @yifuwang noted here: 2/n Consolidate collective functions - collective base and subclasses #9414

Yeah, good point. Will create a separate new feature issue address this